### PR TITLE
Add Typescript support

### PR DESCRIPTION
**Pull request by L4B0MB4**
This is a translation into Typescript. It does **not** improve the overall code quality
- [x] get it running
- [x] tslint completely integrated
- [x] easy start
- [x] easy build
- [x] easy test
- [x] configure webpack to use tslint
- [x] typescript in backend
- [x] typescript in frontend
- [x] configure tslint
- [x] write everything in src/ and backend/ Typescript
- [x] re-add loading screen
#69

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,13 +1,14 @@
 {
   "prettier.endOfLine": "crlf",
   "editor.formatOnSave": true,
-  "editor.wordWrapColumn": 180,
-  "prettier.printWidth": 180,
+  "editor.wordWrapColumn": 120,
```
Comment (L4B0MB4) [inline .vscode/settings.json]: we can change that back to 180. have to do it in the tslint too then

**Comment by L4B0MB4** [comment]
For reviewers: Concentrate on the files that changed as the new files are mostly configuration files for webpack and are copied nearly 100% out of react-typescript ejection

**Comment by L4B0MB4** [comment]
use https://palantir.github.io/tslint/rules/typedef/

**Comment by L4B0MB4** [comment]
@thomasam97 and @N3ol4N : Even if this may not be the final form of the typescript version, I think it would be good to review the changes to get a feeling for typescirpt. As you know how the code looked when it was just javascript, the changes should be not that hard to understand (at least in src/ test/ and backend/ )

File: .travis.yml [hunk 2]
```diff
@@ -2,9 +2,6 @@ language: node_js
 node_js: 11
 cache: npm
 sudo: required
-branches:
```
Comment (L4B0MB4) [inline .travis.yml]: This just re-adds. Branch + PR testing

**Comment by L4B0MB4** [comment]
@thomasam97 I updated the version number. It seems like the homepage field gets used by electron. I find this very weird. But this is the only thing that changed in commit 865a469 and I was able to reproduce it locally.

**Comment by L4B0MB4** [comment]
This pullrequest is done from my side. I´m open for suggestions

**Comment by L4B0MB4** [comment]
Again the weird behavior with the tests. I will fix this. Review anyway pls :)

**Comment by N3ol4N** [comment]
This is a very huge pull request and at this moment in time we lack the knowledge and expertise to do a proper code review.
That's why we did smoke testing and communicated the current bugs and problems in dialoque with the pr-author @L4B0MB4 . We will allow this pr to go through, so the development won't be impaired.
All known bugs will be reported in separat issues and adressed over the next few days/weeks.

**Pull request merged**